go/common/interrupt/interrupt_test.go

```diff
@@ -0,0 +1,32 @@
+package interrupt
+
+import (
+	"context"
+	"syscall"
+	"testing"
+	"time"
+)
+
+func Test_CatchCancelsContextWhenInterrupted(t *testing.T) {
+	ctx := Catch(context.Background())
+	err := syscall.Kill(syscall.Getpid(), syscall.SIGINT)
+	if err != nil {
+		t.Fatal("failed to create a SIGINT signal")
+	}
+	select {
+	case <-time.After(1 * time.Second):
```

Comment:
do not timeout here, it complicates the test. We expect that what is tested works and when it breaks the test will hang, but it will signal that something went wrong.